### PR TITLE
doc: Add detailed info about Bitcoin Core files

### PR DESCRIPTION
**Pull request by hebasto**
This PR:
- provides detailed info about the Bitcoin Core files;
- does not mention temporary files, e.g., `mempool.dat.new` and `peers.????`

### CODE REVIEW

**Review by promag — COMMENTED** [review]
Overall LGTM. Could mention in PR/commit description that references to 0.8.0 are dropped.

**Comment by hebasto** [comment]
> Could mention in PR/commit description that references to 0.8.0 are dropped.

The PR description has been updated.

**Comment by laanwj** [comment]
>  PR/commit description that references to 0.8.0 are dropped.

I think it makes sense to keep a list the "legacy files" somewhere, so that people encountering them (in old installs) have something to search for, and know they are no longer necessary and can be removed. Even if you drop mention of exact versions and meanings.

**Comment by hebasto** [comment]
> I think it makes sense to keep a list the "legacy files" somewhere, so that people encountering them (in old installs) have something to search for, and know they are no longer necessary and can be removed. Even if you drop mention of exact versions and meanings.

Agree. Currently a user can choose any branch from 0.14 to 0.18 in GitHub web interface. Could it be "somewhere" that keeps a list of legacy files?

**Comment by laanwj** [comment]
Please keep it in this document.

**Comment by hebasto** [comment]
> Please keep it in this document.

Done.

**Comment by ch4ot1c** [comment]
Concept ACK.

I like the direction of this @hebasto. Some suggestions:

- `holds`->`contains`
- `custom`->`custom format`, could even link or add information about the format
- `block chain`->`blockchain`
- `bitcoin.conf` could have a reference to the example (`share/examples/bitcoin.conf`)
- `database/` contains data, not `BDB logs`
- Capitalize each line in `Description`
- Using `datadir/` and `wallet_name` as literals is confusing
- `* NOTE:` could be a blockquote or footnote; `regexp`->`regex`
- `** NOTE:` could be a blockquote or footnote, or maybe inlined 
- `blocks/index/`: How do I set a custom path via CLI?

Also, here's another interpretation of the table of contents / section titles, if useful:

```
1. Datadir location
2. Datadir layout
3. Wallet files
  - Default, single wallet layout
  - Multi-wallet layout
4. Other files
  - QSettings
5. Legacy files and directories
```

**Comment by hebasto** [comment]
@ch4ot1c 
>   - `database/` contains data, not `BDB logs`

Considering:
https://github.com/bitcoin/bitcoin/blob/30c2b0b1cb110c27e329e453210561ea5dcdbeb5/src/wallet/db.cpp#L190

and https://docs.oracle.com/cd/E17076_01/html/api_reference/C/envset_lg_dir.html, I believe the contents of the `database/` _are_ BDB logging files.

**Comment by laanwj** [comment]
BDB log files are part of the data though—comparable with the journal in file systems, not as in the debug log.

**Comment by hebasto** [comment]
@ch4ot1c The most part of your suggestions has been implemented.
Didn't catch the last two of them:

> * `blocks/index/`: How do I set a custom path via CLI?
> 
> * Chain option `No option` could spell out the `-mainnet` flag

Would you mind to re-word your ideas?

**Comment by ch4ot1c** [comment]
Sure - 
-  `blocks/index/` - ``Block index; `-blocksdir` option does not affect this path`` - how *do* I affect this path?
- `no option (mainnet)`->``No option (or `-mainnet`)``

**Comment by hebasto** [comment]
@ch4ot1c 
>     * `blocks/index/` - `` Block index; `-blocksdir` option does not affect this path `` - how _do_ I affect this path?

Considering:
https://github.com/bitcoin/bitcoin/blob/94d6a18f23ec1add600f04fc7bd0808b7384d829/src/txdb.cpp#L148-L149

there is no way to affect `blocks/index/` path.

>     * `no option (mainnet)`->`` No option (or `-mainnet`) ``

Do we have `-mainnet` as command-line option?

**Comment by ch4ot1c** [comment]
`-chain=main`, but no. Good notes - thanks for mentioning!

**Comment by ch4ot1c** [comment]
ACK 86b9f92

**Comment by MarcoFalke** [comment]
Concept ACK

File: doc/files.md [hunk 1]
```diff
@@ -1,37 +1,100 @@
...
+`coins/`       | Unspent transaction output database; replaced by `chainstate/` in 0.8.0 | [PR #2231](https://github.com/bitcoin/bitcoin/pull/2231), [`8fdc94cc`](https://github.com/bitcoin/bitcoin/commit/8fdc94cc8f0341e96b1edb3a5b56811c0b20bd15)
+`blkindex.dat` | Blockchain index BDB database; replaced by {`chainstate/`, `blocks/index/`, `blocks/revNNNNN.dat`<sup>[\[2\]](#note2)</sup>} in 0.8.0 | [PR #1677](https://github.com/bitcoin/bitcoin/pull/1677)
+`blk000?.dat`  | Block data (custom format, 2 GiB per file); replaced by `blocks/blkNNNNN.dat`<sup>[\[2\]](#note2)</sup> in 0.8.0 | [PR #1677](https://github.com/bitcoin/bitcoin/pull/1677)
+`addr.dat`     | Peer IP address BDB database; replaced by `peers.dat` in [0.7.0](https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-0.7.0.md) | [PR #1198](https://github.com/bitcoin/bitcoin/pull/1198), [`928d3a01`](https://github.com/bitcoin/bitcoin/commit/928d3a011cc66c7f907c4d053f674ea77dc611cc)
+
+## Notes
+
+<a name="note1">1</a>. The `/` (slash, U+002F) is used as the platform-independent path component separator in this paper.
```
Comment (laanwj) [inline doc/files.md]: lol @ "in this paper"

**Comment by laanwj** [comment]
ACK 86b9f92da25f2768eb29eefd6526320135f39a2f

**Comment by fanquake** [comment]
@hebasto Can you update the PR description. The crossed out `0.8.0` seems incorrect now.

**Comment by hebasto** [comment]
@fanquake 

> Can you update the PR description. The crossed out `0.8.0` seems incorrect now.

Sure ;) Done.

**Comment by MarcoFalke** [comment]
ACK 86b9f92da25f2768eb29eefd6526320135f39a2f

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 86b9f92da25f2768eb29eefd6526320135f39a2f
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUjztQv/cnmPIi2I/WrLg8Tbaj6N4ylQOGWWCt/nMyczT7sjGw1f3x8TP8d9yvi+
s7SJT+IlpBcAZXWsm7RhaCOM9R1K/1dtazMM/ZChKGfVJ+8Ey8KTfSEwayfORkDD
fTBen4typXXyHhAzoyLyTAr35vEH/DoFS8MtBD/Rm48Mo4HgY2DVBfAL8DhpXqVE
AOyJV1GGMKwsAvmoLpJpieSkBzdugxN7x9zwfH9rCKa9F6yUGmLG0Y8W/gMQ4Zfn
5xEHE1BqW9FOzWHWjk32mZN3DsTsbZvfwdaDQ3N/2sYFbcxU/1RT5rCbKdeJIV1l
X2sLXs0yJC7CTa+9OHwlrPpzJtr6Rcua/VpnR4qRJaU+4G0mX0k1mmmaPx1HW4Lt
hOvwuPgOM5QtGnodZI32alYzkLx8qOkbpolPMmYULMS1zGU1IuhIdKO5sUpBJLAo
MdUEGA4xnofNlflcGEY+gkpMAtKaj4cXjLOUnx0mlWXICQL7etdWCYBlaOdNezbx
w8YoInm9
=LO4O
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `9f2ff8ea4a42ee5db2f108e51e215fb6faf8857556cb207b0fe9722786a9906b  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401089f2ff8ea4a42ee5db2f108e51e215fb6faf8857556cb207b0fe9722786a9906bf0101351e96c2a7d6247ebe682c80bd3c73f08fff010826abcae18066c7cd9bf767c0a25eac908f020939f6b70bc6f3bd52f2d7b5af9e9d8b1a61e1f90375f4c9209b957d3d05feb8908f1045d9f8d4bf0084ef5ad39903dc8e50083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010956a5dcf5cd7307a87f57aa63efb219b08f1045d9f8d4bf008755f8bed5d514cd00083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010a7df869b5cd758ab235cb2fc7aee92b608f1045d9f8d4bf0080e274572123272d60083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6df010ddcec5de3eb50a9b228b657505c0f56508f1045d9f8d4bf008e1aac025dacde0c20083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Comment by hebasto** [comment]
@ch4ot1c 

>     * `no option (mainnet)`->`` No option (or `-mainnet`) ``

Oh! I missed that #16680 has been merged recently. It provides `-chain=main` ;)

**Pull request merged**